### PR TITLE
Fixed URL connection client so that it handles error responses

### PR DESCRIPTION
**Pull request by kiiadi**

### CODE REVIEW

File: http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java [hunk 1]
```diff
@@ -132,14 +133,18 @@ public SdkHttpFullResponse call() throws Exception {
...
                                       .statusCode(responseCode)
                                       .statusText(connection.getResponseMessage())
-                                      .content(connection.getInputStream())
+                                      .content(content)
                                       .headers(extractHeaders(connection))
                                       .build();
         }
 
+        private boolean success(int responseCode) {
+            return responseCode < HttpURLConnection.HTTP_BAD_REQUEST;
```
Comment (shorea) [inline http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java]: I'd personally just use 400 here, think it makes it clearer

**Comment by kiiadi** [comment]
I think it's probably worth us having a suite of tests that we apply to any http client implementation to ensure adherence.

**Comment by shorea** [comment]
+1, it is one of our roadmap items although may fall post GA.

**Comment by kiiadi** [comment]
Raised https://github.com/aws/aws-sdk-java-v2/issues/111 for tracking the test suite and changed the success test to use 400 instead of the static argument.

**Pull request merged**